swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/Store.swift

```diff
@@ -94,9 +94,14 @@ public final class Store: ObservableObject {
   }
 
   func createVPNProfile() {
-    DispatchQueue.main.async {
+    DispatchQueue.main.async { [weak self] in
+      guard let self else { return }
+
       Task {
```

Comment:
I don't know Swift very well. What is the point of spawning a task inside a dispatch queue? Those sound like equivalent things?

Reply:
They serve different purposes -- `Task` creates an async context which starts executing immediately, but without any guarantees on which thread.

`DispatchQueue.main` queues the function to execute asynchronously on the main thread, where it's required to be for UI updates. Because the `self.loadTunnelManager` function updates the UI and binds the tunnel manager updates to the MenuBar / UI views, that needs to happen on the main thread.

Reply:
And since the function passed to `DispatchQueue.main.async` must be synchronous, we can't call `try await TunnelManager.shared.create()` directly in that block.

Hmm, actually, since this is the only place where that async function is called, maybe I can just make it blocking. Let me see.